src/winrtble/ble/device.rs

```diff
@@ -58,19 +60,36 @@ impl BLEDevice {
         })
     }
 
-    async fn get_gatt_services(&self) -> Result<GattDeviceServicesResult> {
+    async fn get_gatt_services(
+        &self,
+        cache_mode: BluetoothCacheMode,
+    ) -> Result<GattDeviceServicesResult> {
         let winrt_error = |e| Error::Other(format!("{:?}", e).into());
-        let async_op = self.device.GetGattServicesAsync().map_err(winrt_error)?;
+        let async_op = self
+            .device
+            .GetGattServicesWithCacheModeAsync(cache_mode)
+            .map_err(winrt_error)?;
         let service_result = async_op.await.map_err(winrt_error)?;
         Ok(service_result)
     }
 
     pub async fn connect(&self) -> Result<()> {
-        let service_result = self.get_gatt_services().await?;
+        if self.is_connected().await? {
+            return Ok(());
```

Comment:
Peripheral's `connected` value is never set to true if there already was a connection. It probably should be.

- [ ] set `connected` to true when this returns `Ok(())`?